File: code/iOS/ScriptorSprint2/ViewController.m
//
//  ViewController.m
//  ScriptorSprint2
//
//  Created by Adrian Garcia on 2/20/19.
//  Copyright © 2019 Adrian Garcia. All rights reserved.
//

#import "ViewController.h"
#import <RestKit/RestKit.h>
#import "Songs.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIButton *mainButton;
@property (weak, nonatomic) IBOutlet UIButton *secondaryButton;
@property (weak, nonatomic) IBOutlet UIButton *settingsButton;
@property (weak, nonatomic) IBOutlet UITextView *displayText;

- (void) getJSON;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    [self setFontSize];
}


- (IBAction)mainButtonPressed:(id)sender {
    _mainButton.hidden = true;
    _secondaryButton.hidden = false;
    _displayText.hidden = false;
    [self getJSON];
    [self setFontSize];
    return;
}

- (IBAction)settingsButtonPressed:(id)sender {
    //changes to the settings page
    [self performSegueWithIdentifier:@"toSettings" sender:nil];
    
    
}

- (void) viewDidAppear:(BOOL)animated{
    [self setFontSize];
}


- (IBAction)secondaryButtonPressed:(id)sender {
    [self getJSON];
    [self setFontSize];
    return;
}

- (void) setFontSize {
    NSInteger val = 0;
    NSUserDefaults *settings = [NSUserDefaults standardUserDefaults];
    val = [settings integerForKey:@"fontSize"];
    _displayText.font = [UIFont systemFontOfSize: val];
}


- (void) getJSON {
    NSString *theURL = @"http://35.238.47.23/getData";
    NSURL *connectionURL = [NSURL URLWithString:theURL];
    NSMutableURLRequest *req = [NSMutableURLRequest requestWithURL:connectionURL];
    [req setHTTPMethod:@"GET"];
    NSURLSession *session = [NSURLSession sharedSession];
    NSURLSessionDataTask *task = [session dataTaskWithRequest:req
                                completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
                                    if (error != nil){
                                        NSLog(@"%@", error.localizedDescription);
                                        return;
                                    }
                                    NSData *localData = data;
                                    
                                    NSString *jsonString = [[NSString alloc] initWithData:localData encoding:NSASCIIStringEncoding];
                                    
                                    //jsonString contains the string-ed JSON
                                    NSError *err = nil;
                                    
                                    NSMutableString *toPrint = [NSMutableString stringWithFormat: @""];
                                    [[RKObjectManager sharedManager] getObjectsAtPath:@"/getData" parameters:nil success:^(RKObjectRequestOperation *operation, RKMappingResult *mappingResult) {
                                        
                                        for(NSInteger i = 0; i < mappingResult.array.count; i++){
                                            Songs *song = mappingResult.array[i];
                                            [toPrint appendString: song.mainText];
                                        }
                                        dispatch_async(dispatch_get_main_queue(), ^{
                                            if(toPrint != nil){
                                                self->_displayText.text = toPrint;
                                            } else {
                                                self->_displayText.text = @"An Error Occured...";
                                            }
                                        });
                                        }
                                        failure:^(RKObjectRequestOperation *operation,
                                            NSError *error) {
                                            NSLog(@"What do you mean by 'there is no coffee?': %@", error);
                                        }];
                                    //ends the connection
                                }];
    [task resume];
    return;
    
    
}

@end


//                                                for(int i = 0; i < array.count; i++){
//                                                NSString *temp = [[array objectAtIndex: i] componentsJoinedByString:@""];
//                                                [toPrint appendString:temp];
//                                                }

File: ViewController.swift
//
//  ViewController.swift
//  ScriptorPreAlpha
//
//  Created by Adrian Garcia on 2/16/19.
//  Copyright © 2019 Adrian Garcia. All rights reserved.
//

import UIKit

struct jsonStruct: Decodable {
    var mainText: String
}

class ViewController: UIViewController {

    @IBOutlet weak var displayLabel: UILabel!
    
    @IBOutlet weak var scriptorLabel: UILabel!
    
    @IBOutlet weak var updateButton: UIButton!
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    @IBOutlet weak var mainGetButton: UIButton!
    
    @IBAction func mainGetButtonPressed(_ sender: Any) {
        mainGetButton.isHidden = true
        updateButton.isHidden = false
        getJSON()
        
    }
    @IBAction func updateButtonPressed(_ sender: Any) {
        getJSON()
    
    }
    
    
    func getJSON() -> String {
        let theURL = "http://jowerg.td2s.com:8080/getData"
        let connectionURL = NSURL(string: theURL)
        let req = NSURLRequest(url: connectionURL as! URL)
        var array = [String]()
        URLSession.shared.dataTask(with: req as URLRequest){
            (data, response, error) in
            if(error != nil){
                print(error!.localizedDescription)
                return
            }
            
            let dataLocal = data
            
            let decoder = JSONDecoder()
            var output: [jsonStruct]?
            
            do{
                output = try? decoder.decode([jsonStruct].self, from: dataLocal!)
            }
            var toPrint: String = ""
            for structure in output!{
                
                array.append(structure.mainText)
                toPrint = "\(toPrint)\(structure.mainText)\n"
                
            }
            
            print(toPrint)
            DispatchQueue.main.async(execute: {
                self.displayLabel.text = toPrint
            })
            }.resume()
        return theURL
    }
}

